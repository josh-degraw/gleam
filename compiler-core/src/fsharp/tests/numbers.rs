use crate::assert_fsharp;

#[test]
fn simple_variable_assignment() {
    assert_fsharp!(
        r#"
pub fn main() {
  let a = 3
}
"#
    );
}
