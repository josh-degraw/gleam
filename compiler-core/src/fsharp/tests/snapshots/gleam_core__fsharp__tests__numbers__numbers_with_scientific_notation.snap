---
source: compiler-core/src/fsharp/tests/numbers.rs
assertion_line: 44
expression: "\nconst i = 100.001e523\nconst j = -100.001e-523\n\nconst k = 100.001e1_230\nconst l = -100.001e-1_230\n\nconst m = 100.001e123_456_789\nconst n = -100.001e-123_456_789\n\npub fn main() {\n  i\n  j\n  k\n  l\n  m\n  n\n}\n"
---
module rec my.``mod``
[<Literal>]
let i = 100.001e523
[<Literal>]
let j = -100.001e-523
[<Literal>]
let k = 100.001e1_230
[<Literal>]
let l = -100.001e-1_230
[<Literal>]
let m = 100.001e123_456_789
[<Literal>]
let n = -100.001e-123_456_789
let main(): float = begin
    i
    j
    k
    l
    m
    n
end
